### PR TITLE
graphql inspector settings api

### PR DESCRIPTION
**Pull request by jenshu**
# Description

Add a Settings API to configure GraphQL validation options

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/graphql/v1beta1/stitching_info.proto [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
-package graphql.gloo.solo.io;
-
-option go_package = "github.com/solo-io/gloo/projects/gloo/pkg/api/v1/enterprise/options/graphql/v1beta1";
-
-import "github.com/solo-io/gloo/projects/gloo/api/external/envoy/extensions/graphql/stitching.proto";
-
-// ------------- Graphql Tools JS Input -------------
-// This is not user-facing and is only used to pass to the graphql-tools js script
-// This is the message which the graphql-tools js script will consume
-message GraphQLToolsStitchingInput {
```
Comment (jenshu) [inline projects/gloo/api/v1/enterprise/options/graphql/v1beta1/stitching_info.proto:12]: this file is no longer needed, as it's not user facing and it's been moved to s-p already

File: projects/gloo/api/v1/settings.proto [hunk 2]
```diff
@@ -688,4 +691,38 @@ message ConsoleOptions {
...
+}
+
+// GraphQL settings used by the control plane and UI.
+message GraphqlOptions {
+
+    // Options for how to validate changes to schema definitions.
+    SchemaChangeValidationOptions schema_change_validation_options = 1;
+
+    message SchemaChangeValidationOptions {
+        // Schema definition updates can be considered safe, dangerous, or breaking.
```
Comment (saiskee) [inline projects/gloo/api/v1/settings.proto:703]: Do we want to have any control for blocking dangerous changes as well?

File: projects/gloo/api/v1/settings.proto [hunk 3]
```diff
@@ -688,4 +691,39 @@ message ConsoleOptions {
...
+        // We use [GraphQL Inspector](https://www.graphql-inspector.com/docs/essentials/diff) to detect breaking changes
+        // to GraphQL schemas. This field allows for passing [processing rules](https://www.graphql-inspector.com/docs/essentials/diff#rules)
+        // to GraphQL Inspector to customize how various change types are handled.
+        repeated ProcessingRule processing_rules = 2;
+
+        enum ProcessingRule {
+            RULE_UNSPECIFIED = 0;
+
+            // Turn every dangerous change into a breaking change.
+            RULE_DANGEROUS_BREAKING = 1;
```
Comment (Charlesthebird) [inline projects/gloo/api/v1/settings.proto:717]: Nit: Could this be renamed to something like `RULE_DANGEROUS_TO_BREAKING`? Having it be more descriptive might be useful, although the comment is there.

File: projects/gloo/api/v1/settings.proto [hunk 4]
```diff
@@ -688,4 +691,39 @@ message ConsoleOptions {
...
+
+            // Treat the removal of a deprecated field as a dangerous change, instead of a breaking change.
+            RULE_SUPPRESS_REMOVAL_OF_DEPRECATED_FIELD = 2;
```
Comment (Charlesthebird) [inline projects/gloo/api/v1/settings.proto:720]: Nit: Could this be renamed to something like `RULE_BREAK_ON_REMOVE_DEPRECATED_FIELD`? Suppress could go either way (suppressing the errors on removing deprecated fields, or suppressing changes when removing deprecated fields)

File: projects/gloo/api/v1/settings.proto (hunk 3 above)
Comment (jenshu) [reply projects/gloo/api/v1/settings.proto:717]: sure - i used the same names as graphql-inspector, but we can rename to be more clear

File: projects/gloo/api/v1/settings.proto (hunk 4 above)
Comment (jenshu) [reply projects/gloo/api/v1/settings.proto:720]: hmm, how about something like RULE_REMOVAL_OF_DEPRECATED_FIELD_DANGEROUS?

Comment (Charlesthebird) [reply projects/gloo/api/v1/settings.proto:720]: I think that would be good!

File: projects/gloo/api/v1/settings.proto (hunk 3 above)
Comment (Charlesthebird) [reply projects/gloo/api/v1/settings.proto:717]: That makes sense - I wish their rule names were a little more descriptive 😄  but I think whatever you decide here will be good.

**Review by Charlesthebird — APPROVED** [review]
Looks good!

**Pull request merged**